src/meltano/core/block/extract_load.py

```diff
@@ -609,6 +609,9 @@ async def _link_io(self) -> None:  # noqa: WPS231
                 producer=block.producer,
                 string_id=block.string_id,
                 cmd_type="elb",
+                run_id=str(self.context.job.run_id)
+                if self.context.job
+                else "stateless",
```

Comment:
Not sure about this. Another option is to simply omit adding `run_id` on stateless runs (i.e. those created with `--no-state-update`)